services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js

```diff
@@ -72,13 +73,23 @@ export class MajorMineApplicationPage extends Component {
     a.href = url.url;
   };
 
-  handleSaveData = (e, values, message) => {
+  handleSaveData = async (e, values, message) => {
     e.preventDefault();
     this.props.submit(FORM.ADD_MINE_MAJOR_APPLICATION);
     this.props.touch(FORM.ADD_MINE_MAJOR_APPLICATION);
     const errors = Object.keys(flattenObject(this.props.formErrors));
     if (errors.length === 0) {
-      return this.handleCreateMajorMineApplication(values, message);
+      await this.handleCreateMajorMineApplication(values, message);
```

Comment:
does this not work in the lifecycle hooks? or .then()?

Reply:
.Then() supports it because it is promise based., and it also works in lifecycle methods but may have undesired side effects due to the blocking nature(used sparingly).

I had to use it here because I needed the majorMineApplicationGuid which would only be available after creating a MMA and re-hydrating the store with the fetch.

Reply:
.then() should handle it then but I am not against this either. We have used .then to handle this type of situation in the past but this is the same effect just different implementation.

Reply:
https://nodejs.dev/learn/modern-asynchronous-javascript-with-async-and-await#the-code-is-much-simpler-to-read